src/HealthChecks/HealthChecks/src/DefaultHealthCheckService.cs

```diff
@@ -39,76 +39,81 @@ internal class DefaultHealthCheckService : HealthCheckService
             Func<HealthCheckRegistration, bool> predicate,
             CancellationToken cancellationToken = default)
         {
-            var registrations = _options.Value.Registrations;
-
-            using (var scope = _scopeFactory.CreateScope())
+            async Task<(string registrationName, HealthReportEntry result)> RunCheckAsync(IServiceScope scope, HealthCheckRegistration registration)
```

Comment:
Could this be a regular private method instead? I want to prevent this from capturing things.